floor_generator/lib/processor/entity_processor.dart

```diff
@@ -205,4 +204,24 @@ class EntityProcessor extends QueryableProcessor<Entity> {
 
     return PrimaryKey([primaryKeyField], autoGenerate);
   }
+
+  @nonNull
+  annotations.ForeignKeyAction _getForeignKeyAction(DartObject foreignKeyObject,
+      {@required bool isUpdate}) {
+    final fieldName =
+        isUpdate ? ForeignKeyField.onUpdate : ForeignKeyField.onDelete;
+
+    final field = foreignKeyObject.getField(fieldName);
+    if (field == null) {
+      // field was not defined, return default value
+      return annotations.ForeignKeyAction.noAction;
+    }
+
+    final fka = field.toForeignKeyAction();
+    if (fka == null) {
+      // field was not defined correctly
+      throw _processorError.wrongForeignKeyAction(field, isUpdate);
+    }
+    return fka;
+  }
```

Comment:
💡 We could improve this abstraction by finding a way to not need to pass a boolean to it. We could, for instance, just pass the field name to it instead.